temporalio/worker/_workflow_instance.py

```diff
@@ -1733,6 +1749,7 @@ def _register_task(
 
     def _run_once(self, *, check_conditions: bool) -> None:
         try:
+            self._current_thread_id = get_ident()
             asyncio._set_running_loop(self)
```

Comment:
It seems to me that it's slightly better to set the running loop in `activate()`, and so also set the thread ID there. While we apply activation jobs, it seems that the loop must necessarily remain `self`, right? cc @cretz  Draft PR for that to run CI: https://github.com/temporalio/sdk-python/pull/599

Reply:
Yeah, the original reason is because I didn't want outside-of-workflow-context code (e.g. our code during activate) to think it was in-workflow-context code. I want any expecting-workflow-context code to fail in activate area because it's easy to make a mistake doing too many things not as asyncio tasks.